service/src/main/kotlin/no/nav/su/se/bakover/service/revurdering/RevurderingServiceImpl.kt

```diff
@@ -0,0 +1,140 @@
+package no.nav.su.se.bakover.service.revurdering
+
+import arrow.core.Either
+import arrow.core.getOrElse
+import arrow.core.left
+import arrow.core.right
+import no.nav.su.se.bakover.common.log
+import no.nav.su.se.bakover.common.periode.Periode
+import no.nav.su.se.bakover.database.RevurderingRepo
+import no.nav.su.se.bakover.domain.NavIdentBruker
+import no.nav.su.se.bakover.domain.behandling.Behandling
+import no.nav.su.se.bakover.domain.behandling.BeregnetRevurdering
+import no.nav.su.se.bakover.domain.behandling.OpprettetRevurdering
+import no.nav.su.se.bakover.domain.behandling.Revurdering
+import no.nav.su.se.bakover.domain.behandling.SimulertRevurdering
+import no.nav.su.se.bakover.domain.behandling.TilAttesteringRevurdering
+import no.nav.su.se.bakover.domain.beregning.Beregning
+import no.nav.su.se.bakover.domain.beregning.Beregningsgrunnlag
+import no.nav.su.se.bakover.domain.beregning.fradrag.Fradrag
+import no.nav.su.se.bakover.domain.oppgave.OppgaveConfig
+import no.nav.su.se.bakover.service.oppgave.OppgaveService
+import no.nav.su.se.bakover.service.person.PersonService
+import no.nav.su.se.bakover.service.sak.SakService
+import no.nav.su.se.bakover.service.utbetaling.UtbetalingService
+import java.util.UUID
+
+internal class RevurderingServiceImpl(
+    private val sakService: SakService,
+    private val utbetalingService: UtbetalingService,
+    private val revurderingRepo: RevurderingRepo,
+    private val oppgaveService: OppgaveService,
+    private val personService: PersonService
+) : RevurderingService {
+
+    override fun opprettRevurdering(sakId: UUID, periode: Periode, saksbehandler: NavIdentBruker.Saksbehandler): Either<RevurderingFeilet, Revurdering> {
+        // TODO logikk for å finne ut hva som skal revurderes
+        return hentSak(sakId)
+            .map { sak ->
+                val tilRevurdering = sak.behandlinger()
+                    .filter { it.status() == Behandling.BehandlingsStatus.IVERKSATT_INNVILGET }
+                    .firstOrNull() { it.beregning()!!.getPeriode() inneholder periode }
+                return when (tilRevurdering) {
+                    null -> RevurderingFeilet.FantIngentingSomKanRevurderes.left()
+                    else -> {
+                        val revurdering = OpprettetRevurdering(tilRevurdering = tilRevurdering, saksbehandler = saksbehandler)
+                        revurderingRepo.lagre(revurdering)
+                        revurderingRepo.hent(revurdering.id)!!.right()
+                    }
+                }
+            }
+    }
+
+    override fun beregnOgSimuler(
+        revurderingId: UUID,
+        saksbehandler: NavIdentBruker.Saksbehandler,
+        periode: Periode,
```

Comment:
Kommentere også tidligere at det er litt rart å kunne overstyre perioden man satt i opprettet? Da vil den kunne gjelde for andre behandlinger.